fili-core/src/main/java/com/yahoo/bard/webservice/web/DataApiRequest.java

```diff
@@ -200,40 +207,70 @@ public DataApiRequest(
 
         DateTimeFormatter dateTimeFormatter = generateDateTimeFormatter(timeZone);
 
-        this.intervals = generateIntervals(intervals, this.granularity, dateTimeFormatter);
-
-        metricDictionary = metricDictionary.getScope(Collections.singletonList(tableName));
+        this.intervals = timeGenerator(
+                () -> generateIntervals(intervals, this.granularity, dateTimeFormatter),
+                "GeneratingIntervals"
+        );
 
         this.filterBuilder = bardConfigResources.getFilterBuilder();
 
+        MetricDictionary metricDictionary = bardConfigResources
+                .getMetricDictionary()
+                .getScope(Collections.singletonList(tableName));
```

Comment:
This change was necessary to make `metricDictionary` effectively final for use in lambdas (also the previous approach was confusing).